include/nbl/builtin/hlsl/sort/counting.hlsl

```diff
@@ -79,30 +81,36 @@ struct counting
         sdata.workgroupExecutionAndMemoryBarrier();
 
         uint32_t sum = inclusive_scan(histogram_value, sdata);
-        toroidal_histogram_add(tid, sum, histogram, sdata, params);
+        histogram.atomicAdd(tid, sum);
 
         const bool is_last_wg_invocation = tid == (GroupSize - 1);
         const uint16_t adjusted_key_bucket_count = ((KeyBucketCount - 1) / GroupSize + 1) * GroupSize;
 
-        for (tid += GroupSize; tid < adjusted_key_bucket_count; tid += GroupSize)
+        for (uint32_t vid = tid + GroupSize; vid < adjusted_key_bucket_count; vid += GroupSize)
         {
             if (is_last_wg_invocation)
             {
-                uint32_t startIndex = tid - tid % GroupSize;
+                uint32_t startIndex = vid - tid;
                 sdata.set(startIndex, sdata.get(startIndex) + sum);
             }
 
-            sum = inclusive_scan(sdata.get(tid), sdata);
-            toroidal_histogram_add(tid, sum, histogram, sdata, params);
+            sum = inclusive_scan(sdata.get(vid), sdata);
+            histogram.atomicAdd(vid, sum);
```

Comment:
you need to guard the `atomicAdd` with `vid<KeyBucketCount` otherwise you do OOB access

Reply:
also how you get the `sdata.get(vid)` https://github.com/Devsh-Graphics-Programming/Nabla/pull/690#discussion_r1621424265